ui/src/components/WorkerDetailsCard/index.jsx

```diff
@@ -1,52 +1,189 @@
-import React, { Component } from 'react';
+import React, { Component, Fragment } from 'react';
 import { format, parseISO } from 'date-fns';
-import List from '@material-ui/core/List';
-import ListItem from '@material-ui/core/ListItem';
-import ListItemText from '@material-ui/core/ListItemText';
+import {
+  Button,
+  List,
+  ListItem,
+  ListItemText,
+  Typography,
+} from '@material-ui/core';
+import DeleteIcon from 'mdi-react/DeleteIcon';
+import { WorkerManager } from 'taskcluster-client-web';
 import DateDistance from '../DateDistance';
+import DialogAction from '../DialogAction';
+import Label from '../Label';
+import { withAuth } from '../../utils/Auth';
 import { worker } from '../../utils/prop-types';
+import { NULL_PROVIDER } from '../../utils/constants';
 
 /**
  * Render information in a card layout about a worker.
  */
+@withAuth
 export default class WorkerDetailsCard extends Component {
   static propTypes = {
     /** A GraphQL worker response. */
     worker: worker.isRequired,
   };
 
+  state = {
```

Comment:
iirc, nested objects inside state are not so common, and usually state is a plain object.
Since you don't have anything else there besides `dialogState`, maybe it can all go to the top level?